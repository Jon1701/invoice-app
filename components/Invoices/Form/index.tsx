import React from 'react';
import styled from 'styled-components';

import {
  Action,
  setCurrencyCode,
  setBillerName,
  setBillerEmail,
  setBillerAddressLine1,
  setBillerAddressLine2,
  setBillerAddressCity,
  setBillerAddressState,
  setBillerAddressPostalCode,
  setBillerAddressCountry,
  setClientName,
  setClientEmail,
  setClientAddressLine1,
  setClientAddressLine2,
  setClientAddressCity,
  setClientAddressState,
  setClientAddressPostalCode,
  setClientAddressCountry,
  addBlankInvoiceItem,
} from '@components/Invoices/Form/actions';
import { blankInvoice } from '@components/Invoices/Form';
import { Invoice, CurrencyCode } from '@appTypes/index';
import { Button, ColorScheme, Shape, Variant } from '@components/Button';
import { breakpoints } from '@utils/breakpoints';
import Dropdown from '@components/Form/Dropdown';
import Label from '@components/Form/Label';

import { Legend } from './subcomponents/styles';
import AddItemButton from './subcomponents/AddItemButton';
import ContactFieldset from './subcomponents/ContactFieldset';
import DisplayInvoiceItems from './subcomponents/DisplayInvoiceItems';

/**
 * Styled form.
 */
const Form = styled.form`
  & > fieldset:not(:last-child) {
    margin-bottom: 15px;
  }
`;

/**
 * Styled fieldset.
 */
const Fieldset = styled.fieldset`
  all: unset;
  display: block;

  & > *:not(legend) {
    margin-bottom: 15px;
  }
`;

/**
 * Container for the Submit/Cancel buttons.
 */
const ContainerButtons = styled.div`
  display: grid;
  gap: 15px;

  @media screen and (max-width: ${breakpoints.mobile.max}px) {
    grid-template-rows: repeat(2, 1fr);
    grid-template-columns: repeat(1, 1fr);
  }

  @media screen and (min-width: ${breakpoints.tablet.min}px) {
    grid-template-rows: repeat(1, 1fr);
    grid-template-columns: repeat(2, 250px);

    justify-content: center;
  }
`;

/**
 * Form props.
 */
export interface InvoiceFormProps<ActionCreatorType> {
  /**
   * Submission handler.
   */
  handleSubmit?: React.FormEventHandler<HTMLFormElement>;

  /**
   * Resets form state.
   */
  handleReset?: React.MouseEventHandler<HTMLButtonElement>;

  /**
   * Form values.
   */
  formValues: Invoice;

  /**
   * Dispatches actions.
   */
  dispatch: React.Dispatch<ActionCreatorType>;

  /**
   * Indicates if the form is read-only.
   */
  readOnly?: boolean;

  /**
   * Indicates if the form is disabled.
   */
  disabled?: boolean;
}

/**
 * Form to Create an Invoice.
 */
export const InvoiceForm: React.FC<InvoiceFormProps<Action>> = ({
  handleSubmit,
  handleReset,
  dispatch,
  formValues,
  readOnly = false,
  disabled = false,
}) => {
  return (
    <Form onSubmit={handleSubmit}>
      <ContactFieldset
        title="Bill From"
        contactInfo={formValues.biller}
        dispatch={dispatch}
        setName={setBillerName}
        setEmail={setBillerEmail}
        setAddressLine1={setBillerAddressLine1}
        setAddressLine2={setBillerAddressLine2}
        setAddressCity={setBillerAddressCity}
        setAddressState={setBillerAddressState}
        setAddressPostalCode={setBillerAddressPostalCode}
        setAddressCountry={setBillerAddressCountry}
        readOnly={readOnly}
        disabled={disabled}
      />

      <ContactFieldset
        title="Client"
        contactInfo={formValues.client}
        dispatch={dispatch}
        setName={setClientName}
        setEmail={setClientEmail}
        setAddressLine1={setClientAddressLine1}
        setAddressLine2={setClientAddressLine2}
        setAddressCity={setClientAddressCity}
        setAddressState={setClientAddressState}
        setAddressPostalCode={setClientAddressPostalCode}
        setAddressCountry={setClientAddressCountry}
        readOnly={readOnly}
        disabled={disabled}
      />

      <Fieldset>
        <Legend>Items</Legend>

        <div>
          <Label htmlFor="currency-code" isRequired>
            Currency
          </Label>
          <Dropdown
            id="currency-code"
            handleChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
              dispatch(setCurrencyCode(e.target.value as CurrencyCode));
            }}
            value={formValues.currencyCode}
            disabled={disabled || readOnly}
            handleClearButtonClick={() =>
              dispatch(setCurrencyCode(blankInvoice.currencyCode))
            }>
            <option disabled hidden>
              (Select one)
            </option>
            <option value={CurrencyCode.CAD}>Canadian Dollar (CAD)</option>
            <option value={CurrencyCode.USD}>United States Dollar (USD)</option>
          </Dropdown>
        </div>

        {formValues?.items.length > 0 ? (
          <DisplayInvoiceItems
            currencyCode={formValues.currencyCode}
            items={formValues.items}
          />
        ) : undefined}

        {!(readOnly || disabled) ? (
          <AddItemButton
            handleClick={() => {
              dispatch(addBlankInvoiceItem());
            }}
          />
        ) : undefined}

        <ContainerButtons>
          <Button
            type="submit"
            shape={Shape.Rounded}
            variant={Variant.Solid}
            colorScheme={ColorScheme.Purple}>
            Submit
          </Button>
          {handleReset !== undefined ? (
            <Button
              type="reset"
              onClick={handleReset}
              shape={Shape.Rounded}
              variant={Variant.Hollow}
              colorScheme={ColorScheme.Purple}>
              Clear
            </Button>
          ) : undefined}
        </ContainerButtons>
      </Fieldset>
    </Form>
  );
};

export default InvoiceForm;

// Re-export modules from actions.
export * from './actions';

// Re-export modules from reducers.
export * from './reducers';
