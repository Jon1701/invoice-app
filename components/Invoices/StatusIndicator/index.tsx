import React from 'react';
import styled, { StyledComponent } from 'styled-components';
import { Invoice, InvoiceStatus } from '@appTypes/index';

/**
 * Indicator default style.
 */
const StyledBaseIndicator = styled.span`
  border-style: solid;
  border-width: 1px;
  border-color: #084298;
  font-size: 0.8em;

  background-color: #031633;
  color: #6ea8fe;

  padding: 2px 25px;
`;

/**
 * Draft indicator.
 */
const StyledDraftIndicator = styled(StyledBaseIndicator)`
  border-color: #495057;
  background-color: #343a40;
  color: #f8f9fa;
`;

/**
 * Pending indicator.
 */
const StyledPendingIndicator = styled(StyledBaseIndicator)`
  border-color: #664d03;
  background-color: #332701;
  color: #ffda6a;
`;

/**
 * Paid indicator.
 */
const StyledPaidIndicator = styled(StyledBaseIndicator)`
  border-color: #0f5132;
  background-color: #051b11;
  color: #75b798;
`;

/**
 * Returns the user-friendly status from an Invoice status enum.
 */
export const displayStatus = (status: InvoiceStatus): string => {
  switch (status) {
    case InvoiceStatus.Draft:
      return 'Draft';

    case InvoiceStatus.Paid:
      return 'Paid';

    case InvoiceStatus.Pending:
      return 'Pending';

    default:
      return 'Unknown';
  }
};

interface Props {
  /**
   * Invoice status.
   */
  status: InvoiceStatus;
}

/**
 * Returns a user-friendly status indicator component from an Invoice
 * status enum.
 */
export const StatusIndicator: React.FC<Props> = ({ status }: Props) => {
  let Container = StyledBaseIndicator;
  const text = displayStatus(status);

  switch (status) {
    case InvoiceStatus.Draft:
      Container = StyledDraftIndicator;
      break;

    case InvoiceStatus.Pending:
      Container = StyledPendingIndicator;
      break;

    case InvoiceStatus.Paid:
      Container = StyledPaidIndicator;
      break;

    default:
      Container = StyledBaseIndicator;
      break;
  }

  return <Container>{text}</Container>;
};
