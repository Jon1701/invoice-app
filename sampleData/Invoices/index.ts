import { Currency, Invoice, InvoiceStatus } from '@appTypes/index';
import calculateInvoiceTotalFromItems from '@utils/invoices/calculateInvoiceTotalFromItems';

// Original Invoice data with placeholder calculated values to satisfy interface
// requirements.
const originalData: Array<Invoice> = [
  {
    id: '1',
    amount: { total: 0 },
    currency: Currency.CAD,
    status: InvoiceStatus.Draft,
    biller: {
      name: 'Jonathan Archer',
      email: 'jarcher@nx01.com',
      address: {
        line1: '01 Enterprise Avenue',
        line2: "Captain's Quarters",
        city: 'San Francisco',
        state: 'California',
        postalCode: '12345',
        country: 'United States',
      },
    },
    client: {
      name: 'Kathryn Janeway',
      email: 'kathryn.janeway@ncc74656.com',
      address: {
        line1: '74656 Voyager Path',
        city: 'Calgary',
        state: 'Alberta',
        postalCode: 'L1H3C1',
        country: 'Canada',
      },
    },
    items: [
      {
        id: 'invoiceitem_1a881a28d41345059705ea73209a3447',
        description: 'Self Sealing Stem Bolt',
        unitPrice: 36.18,
        quantity: 5,
      },
      {
        id: 'invoiceitem_68450b8e-8276-4dc7-b720-91b07fadec85',
        description: 'Reverse Ratcheting Router',
        unitPrice: 4.32,
        quantity: 10,
      },
      {
        id: 'invoiceitem_60e546e905b2422ba945ca7911fe7f50',
        description: 'Gold-pressed Latinum',
        unitPrice: 12.36,
        quantity: 88,
      },
      {
        id: 'invoiceitem_a40957b920af41f08ea89e9c192b2076',
        description: 'Plasma Conduit',
        unitPrice: 9102,
        quantity: 84,
      },
      {
        id: 'invoiceitem_a6be1a51b249453abd2f301bd1b204d6',
        description: 'Quantum Torpedo',
        unitPrice: 123957,
        quantity: 47,
      },
      {
        id: 'invoiceitem_3007759a86c34937a41e5b18d49c42e6',
        description: 'Medical Tricorder',
        unitPrice: 2582,
        quantity: 155,
      },
      {
        id: 'invoiceitem_4c461cf74b3a4b90a961b19e3631842b',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
        unitPrice: 120346879,
        quantity: 1325,
      },
      {
        id: 'invoiceitem_9e257977e9364b82b4422d2dd2d0a0aa',
        description: 'Polarized Hull Plate',
        unitPrice: 356812,
        quantity: 1825,
      },
      {
        id: 'invoiceitem_bfb9bc3326cb4a87a163ce56c2601067',
        description: 'Antimatter Injector',
        unitPrice: 47387,
        quantity: 5,
      },
    ],
  },
  {
    id: '2',
    amount: { total: 0 },
    currency: Currency.USD,
    status: InvoiceStatus.Pending,
    biller: {
      name: 'Hikaru Sulu',
      email: 'sulu.hikaru@excelsior.net',
      address: {
        line1: '2000 Excelsior Road',
        line2: 'N/A',
        city: 'San Francisco',
        state: 'California',
        postalCode: '12856',
        country: 'United States',
      },
    },
    client: {
      name: 'James Kirk',
      email: 'jtk@enterprise.com',
      address: {
        line1: '1701 Enterprise Road',
        line2: 'Block A',
        city: 'Toronto',
        state: 'Ontario',
        postalCode: 'K8S7H6',
        country: 'Canada',
      },
    },
    items: [
      {
        id: 'invoiceitem_0422e201d4514d1b912ba74ef3860e14',
        description: 'Tricorder',
        unitPrice: 3617.12,
        quantity: 277,
      },
      {
        id: 'invoiceitem_8bf64b9dafb94a3a9b2ed741c1e64cfe',
        description: 'Medical Tricorder',
        unitPrice: 5.84,
        quantity: 3693,
      },
      {
        id: 'invoiceitem_6b2ec443bdde405eb735e2cbee65a361',
        description: 'Tribble',
        unitPrice: 7.9,
        quantity: 363,
      },
      {
        id: 'invoiceitem_1deee6c0df3c4a058f717bea813a2c0b',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Augue eget arcu dictum varius duis at consectetur lorem donec. Egestas sed tempus urna et pharetra pharetra massa. Bibendum enim facilisis gravida neque convallis a cras. Nunc aliquet bibendum enim facilisis gravida neque convallis. Lacus vestibulum sed arcu non odio euismod lacinia at. Varius sit amet mattis vulputate. Orci dapibus ultrices in iaculis. A erat nam at lectus urna duis convallis convallis. Vel eros donec ac odio tempor orci dapibus ultrices. Metus vulputate eu scelerisque felis imperdiet proin. Dictum non consectetur a erat nam at lectus urna. Dictum sit amet justo donec enim diam vulputate ut pharetra.',
        unitPrice: 25929578.3,
        quantity: 346234,
      },
    ],
  },
  {
    id: '3',
    amount: { total: 0 },
    currency: Currency.USD,
    status: InvoiceStatus.Pending,
    biller: {
      name: 'Jean-Luc Picard',
      email: 'jlp@stargazer.com',
      address: {
        line1: '2893 Stargazer Lane',
        line2: 'Suite AC3A',
        city: 'Winnipeg',
        state: 'Manitoba',
        postalCode: 'K8B2V1',
        country: 'Canada',
      },
    },
    client: {
      name: 'Locutus of Borg',
      email: 'locutus@borg.com',
      address: {
        line1: '634 Sphere Street',
        city: 'Toronto',
        state: 'Ontario',
        postalCode: 'J8M2G1',
        country: 'Canada',
      },
    },
    items: [
      {
        id: 'invoiceitem_a43047d100d8459198fbfb59bdfd9f8e',
        description: 'M-5 Computer',
        unitPrice: 3623441,
        quantity: 15,
      },
    ],
  },
  {
    id: '4',
    amount: { total: 0 },
    currency: Currency.CAD,
    status: InvoiceStatus.Paid,
    biller: {
      name: 'William T. Riker',
      email: 'riker@titan.com',
      address: {
        line1: '80102 Titan Avenue',
        city: 'Unknown City',
        state: 'Unknown State',
        postalCode: 'M7B6D1',
        country: 'Canada',
      },
    },
    client: {
      name: 'Beverly Crusher',
      email: 'crusher@sf.medical',
      address: {
        line1: '18276 Pasteur Lane',
        city: 'Los Angeles',
        state: 'California',
        postalCode: '89182',
        country: 'United States',
      },
    },
    items: [],
  },
];

// Transformed data with calculated values.
export const sampleInvoices: Array<Invoice> = originalData.map(
  (invoice: Invoice) => {
    // Get total amount from items (quantity and unit price.)
    const total: number = invoice.items.reduce(
      calculateInvoiceTotalFromItems,
      0
    );

    invoice.amount.total = total;

    return invoice;
  }
);
