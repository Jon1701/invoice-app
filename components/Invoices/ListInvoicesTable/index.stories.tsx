import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Currency, Invoice, InvoiceStatus } from '@appTypes/index';
import { sampleInvoices } from '@sampleData/Invoices';

import ListInvoicesTable from '.';

export default {
  component: ListInvoicesTable,
} as ComponentMeta<typeof ListInvoicesTable>;

const Template: ComponentStory<typeof ListInvoicesTable> = args => (
  <ListInvoicesTable {...args} />
);

export const TableWithData = Template.bind({});
TableWithData.args = {
  invoices: sampleInvoices,
};

export const TableWithNoData = Template.bind({});
TableWithNoData.args = {
  invoices: [],
};
