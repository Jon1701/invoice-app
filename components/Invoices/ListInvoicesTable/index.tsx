import React from 'react';
import styled from 'styled-components';

import { Invoice, InvoiceItem } from '@appTypes/index';

import { DisplayFormattedCurrency } from '@components/Currency';
import { StatusIndicator } from '@components/Invoices/StatusIndicator';

/**
 * Styled table.
 */
const Table = styled.table`
  width: 100%;
  border-collapse: separate;
  border-spacing: 0 5px;
  text-align: center;
`;

/**
 * Styled table heading.
 */
const Heading = styled.th`
  padding: 15px;
`;

/**
 * Styled table row.
 */
const Row = styled.tr`
  background-color: #453c67;
`;

/**
 * Styled table column.
 */
const Column = styled.td`
  padding: 15px;
`;

/**
 * Container for the No Invoices message.
 */
const ContainerNoInvoices = styled.div`
  background-color: #453c67;
  text-align: center;

  padding: 15px;
`;

interface Props {
  /**
   * Array of Invoices.
   */
  invoices: Array<Invoice>;
}

/**
 * Displays a list of Invoices.
 */
const ListInvoicesTable: React.FC<Props> = ({ invoices }: Props) => {
  if (invoices.length === 0) {
    return <ContainerNoInvoices>No Invoices found.</ContainerNoInvoices>;
  }

  const nodes = invoices.map((invoice: Invoice) => {
    return (
      <Row key={invoice.id}>
        <Column>{invoice.id}</Column>
        <Column>
          <StatusIndicator status={invoice.status} />
        </Column>
        <Column>{invoice.client.name}</Column>
        <Column>
          <DisplayFormattedCurrency
            currencyCode={invoice.currency}
            amount={invoice.amount.total}
          />
        </Column>
      </Row>
    );
  });

  return (
    <Table>
      <thead>
        <Row>
          <Heading>ID</Heading>
          <Heading>Status</Heading>
          <Heading>Client Name</Heading>
          <Heading>Amount</Heading>
        </Row>
      </thead>
      <tbody>{nodes}</tbody>
    </Table>
  );
};

export default ListInvoicesTable;
