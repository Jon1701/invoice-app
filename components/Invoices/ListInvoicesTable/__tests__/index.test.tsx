import React from 'react';
import { describe, test } from '@jest/globals';
import { render } from '@testing-library/react';

import { Invoice } from '@appTypes/index';
import { sampleInvoices } from '@sampleData/Invoices';

import ListInvoicesTable from '..';

describe('<ListInvoicesTable />', () => {
  describe('with empty invoices prop', () => {
    const invoices: Array<Invoice> = [];

    test('it should render without errors', () => {
      const node = render(<ListInvoicesTable invoices={invoices} />);

      expect(node).toMatchSnapshot();
    });
  });

  describe('with prop invoices set to sample data', () => {
    const invoices: Array<Invoice> = sampleInvoices;

    test('it should render without errors', () => {
      const node = render(<ListInvoicesTable invoices={invoices} />);

      expect(node).toMatchSnapshot();
    });
  });
});
