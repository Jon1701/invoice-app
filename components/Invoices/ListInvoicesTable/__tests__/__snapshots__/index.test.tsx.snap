// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListInvoicesTable /> with empty invoices prop it should render without errors 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ftTHYK kpkkbv"
      >
        No invoices found.
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ftTHYK kpkkbv"
    >
      No invoices found.
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ListInvoicesTable /> with prop invoices set to sample data it should render without errors 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="sc-eDvSVe efzZSJ"
      >
        <tr
          class="sc-gKPRtg iZLckb"
        >
          <th
            class="sc-jSUZER ebMnHH"
          >
            ID
          </th>
          <th
            class="sc-jSUZER ebMnHH"
          >
            Status
          </th>
          <th
            class="sc-jSUZER ebMnHH"
          >
            Client Name
          </th>
          <th
            class="sc-jSUZER ebMnHH"
          >
            Amount
          </th>
        </tr>
        <tr
          class="sc-gKPRtg iZLckb"
        >
          <td
            class="sc-iBYQkv dUNxtU"
          >
            1
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            <span
              class="sc-bcXHqe sc-gswNZR grKKXY gVoFOn"
            >
              Draft
            </span>
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            Jonathan Archer
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            <span>
              $131,178.00
            </span>
          </td>
        </tr>
        <tr
          class="sc-gKPRtg iZLckb"
        >
          <td
            class="sc-iBYQkv dUNxtU"
          >
            2
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            <span
              class="sc-bcXHqe sc-dkrFOg grKKXY hRcOVh"
            >
              Pending
            </span>
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            Hikaru Sulu
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            <span>
              $3,445,391.00
            </span>
          </td>
        </tr>
        <tr
          class="sc-gKPRtg iZLckb"
        >
          <td
            class="sc-iBYQkv dUNxtU"
          >
            3
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            <span
              class="sc-bcXHqe sc-dkrFOg grKKXY hRcOVh"
            >
              Pending
            </span>
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            Jean-Luc Picard
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            <span>
              ¥292,583,700
            </span>
          </td>
        </tr>
        <tr
          class="sc-gKPRtg iZLckb"
        >
          <td
            class="sc-iBYQkv dUNxtU"
          >
            4
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            <span
              class="sc-bcXHqe sc-hLBbgP grKKXY divHvX"
            >
              Paid
            </span>
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            William T. Riker
          </td>
          <td
            class="sc-iBYQkv dUNxtU"
          >
            <span>
              $0.00
            </span>
          </td>
        </tr>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="sc-eDvSVe efzZSJ"
    >
      <tr
        class="sc-gKPRtg iZLckb"
      >
        <th
          class="sc-jSUZER ebMnHH"
        >
          ID
        </th>
        <th
          class="sc-jSUZER ebMnHH"
        >
          Status
        </th>
        <th
          class="sc-jSUZER ebMnHH"
        >
          Client Name
        </th>
        <th
          class="sc-jSUZER ebMnHH"
        >
          Amount
        </th>
      </tr>
      <tr
        class="sc-gKPRtg iZLckb"
      >
        <td
          class="sc-iBYQkv dUNxtU"
        >
          1
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          <span
            class="sc-bcXHqe sc-gswNZR grKKXY gVoFOn"
          >
            Draft
          </span>
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          Jonathan Archer
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          <span>
            $131,178.00
          </span>
        </td>
      </tr>
      <tr
        class="sc-gKPRtg iZLckb"
      >
        <td
          class="sc-iBYQkv dUNxtU"
        >
          2
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          <span
            class="sc-bcXHqe sc-dkrFOg grKKXY hRcOVh"
          >
            Pending
          </span>
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          Hikaru Sulu
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          <span>
            $3,445,391.00
          </span>
        </td>
      </tr>
      <tr
        class="sc-gKPRtg iZLckb"
      >
        <td
          class="sc-iBYQkv dUNxtU"
        >
          3
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          <span
            class="sc-bcXHqe sc-dkrFOg grKKXY hRcOVh"
          >
            Pending
          </span>
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          Jean-Luc Picard
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          <span>
            ¥292,583,700
          </span>
        </td>
      </tr>
      <tr
        class="sc-gKPRtg iZLckb"
      >
        <td
          class="sc-iBYQkv dUNxtU"
        >
          4
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          <span
            class="sc-bcXHqe sc-hLBbgP grKKXY divHvX"
          >
            Paid
          </span>
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          William T. Riker
        </td>
        <td
          class="sc-iBYQkv dUNxtU"
        >
          <span>
            $0.00
          </span>
        </td>
      </tr>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
